packages/tools/src/tools/ScaleOverlayTool.ts

```diff
@@ -0,0 +1,666 @@
+    };
+  };
+
+  onSetToolEnabled = (): void => {
+    this._init();
+  };
+
+  onCameraModified = (evt: Types.EventTypes.CameraModifiedEvent): void => {
+    // If the camera is modified, we need to update the viewport
+    // that the camera was modified on
+    this.configuration.viewportId = evt.detail.viewportId;
+    this._init();
+  };
+
+  /**
+   * Used to draw the scale annotation in each request animation
+   * frame.
+   *
+   * @param enabledElement - The Cornerstone's enabledElement.
+   * @param svgDrawingHelper - The svgDrawingHelper providing the context for drawing.
+   * @returns
+   */
+
+  renderAnnotation(
+    enabledElement: Types.IEnabledElement,
+    svgDrawingHelper: SVGDrawingHelper
+  ) {
+    if (!this.editData.viewport) {
+      return;
+    }
+    const location = this.configuration.scaleLocation;
+    const { annotation, viewport } = this.editData;
+    const image = viewport.getImageData();
+    const imageId = viewport.getCurrentImageId();
+    const canvas = enabledElement.viewport.canvas;
+
+    const renderStatus = false;
+
+    if (!viewport) {
+      return renderStatus;
+    }
+
+    const styleSpecifier: StyleSpecifier = {
+      toolGroupId: this.toolGroupId,
+      toolName: this.getToolName(),
+      viewportId: enabledElement.viewport.id,
+    };
+
+    let rowPixelSpacing = image.spacing[0];
+    let colPixelSpacing = image.spacing[1];
+    const imagePlane = metaData.get('imagePlaneModule', imageId);
+
+    // if imagePlane exists, set row and col pixel spacing
+    if (imagePlane) {
+      rowPixelSpacing =
+        imagePlane.rowPixelSpacing || imagePlane.rowImagePixelSpacing;
+      colPixelSpacing =
+        imagePlane.columnPixelSpacing || imagePlane.colImagePixelSpacing;
+    }
+
+    // Check whether pixel spacing is defined
+    if (!rowPixelSpacing || !colPixelSpacing) {
+      console.warn(
+        `Unable to define rowPixelSpacing or colPixelSpacing from data on ScaleOverlayTool's renderAnnotation`
+      );
+      return;
+    }
+
+    const canvasSize = {
+      width: canvas.width,
+      height: canvas.height,
+    };
+
+    const topLeft = annotation.data.handles.points[0];
+    const topRight = annotation.data.handles.points[1];
+    const bottomLeft = annotation.data.handles.points[2];
+    const bottomRight = annotation.data.handles.points[3];
+
+    const pointSet1 = [topLeft, bottomLeft, topRight, bottomRight];
+
+    const worldWidthViewport = vec3.distance(bottomLeft, bottomRight);
+    const worldHeightViewport = vec3.distance(topLeft, bottomLeft);
+
+    // 0.05 gives margin to horizontal and vertical lines
+    const hscaleBounds = this.computeScaleBounds(
+      canvasSize,
+      0.05,
+      0.05,
+      location
+    );
+
+    const vscaleBounds = this.computeScaleBounds(
+      canvasSize,
+      0.05,
+      0.05,
+      location
+    );
+
+    const scaleSize = this.computeScaleSize(
+      worldWidthViewport,
+      worldHeightViewport,
+      location
+    );
```

Comment:
These require documentation as a comment above each. Not sure what each function is doing

Reply:
Added documentation